### PR TITLE
Support Sandbox Image on macOS

### LINKED ISSUE #2871: MacOS alpha version does not work with sandbox images
## Version of Singularity:

First alpha release of MacOS version of singularity.  singularity --version says 3.1.0-196.g8d47ee66.dirty

### Expected behavior

Works with sandbox images.

### Actual behavior

FATAL:   image format not recognized

### Steps to reproduce behavior

Copy a sandbox image from a Linux machine and do singularity shell sandbox_dirname

@jmstover

### PR DESCRIPTION
**Pull request by jmstover**
Check if we have a directory used for the image, and if so be sure to pass it instead of a image.

This will help support running a Sandbox image from a macOS host.

**This fixes or addresses the following GitHub issues:**

- Fixes #2871 

Attn: @singularity-maintainers

### CODE REVIEW

File: cmd/internal/cli/startvm_darwin.go [hunk 1]
```diff
@@ -70,16 +69,31 @@ func getHypervisorArgs(sifImage, bzImage, initramfs, singAction, cliExtra string
...
 	args = append(args, "-s", pciArgs)
 
+	// Check for Sandbox Image
+	sylog.Debugf("Check for sandbox image")
+	if f, err := os.Stat(sifImage); err == nil {
+		if f.IsDir() {
+			sylog.Debugf("Image is sandbox. Setting up share.")
+			pciArgs = fmt.Sprintf("4:1,virtio-9p,runimg=%s", sifImage)
+			args = append(args, "-s", pciArgs)
+			sboxImgBind := fmt.Sprintf("runimg:/runImage")
```
Comment (cclerget) [inline cmd/internal/cli/startvm_darwin.go:89]: `runImage` or sifImage value ?

Comment (jmstover) [reply cmd/internal/cli/startvm_darwin.go:89]: `runimg` is the 9p mount tag. `/runImage` is the mount point.

Comment (jmstover) [reply cmd/internal/cli/startvm_darwin.go:89]: To expand on this... line 87, it sets:
`virtio-9p,runimg=[sifImage]`

Where `runimg` is the 9p mount tag, and what we're passing to be mounted is the value of `sifImage`, which in this case is a sandbox directory.

**Review by cclerget — APPROVED** [review]
LGTM

**Pull request merged**